### ISSUE TITLE
[allow_hotloading_data] update_cache needed to register new country

### ISSUE THREAD

**Issue by clm-a** [hexorx/countries#353]
Hi there !
FYI, on the branch **allow_hotloading_data** I have seen that registering new data in a plain Rails initializer kills the normal data cache.

I'm using the **country_select** / countries2 branch fork from @scudco I get the issue even if I disable the country_select gem.

PS : Canary Islands is not an ISO3166 entry but I need to make the disctinction for shipping cost purposes.

```ruby
# this code will only make IC available for ISO3166::Country.new --- @@cache is empty before register call and @@registered_data contains IC afterwards
ISO3166::Data.register(
  alpha2: "IC",
  name: 'Spain - Canary Islands',
  translations: {
    'EN' => "Spain - Canary Islands",
    'DE' => "Spanien - Kanarische Inseln",
    'NL' => "Spanje - Canarische Eilanden",
    'JA' => "スペイン - カナリア諸島",
    'ES' => "España - Islas Canarias",
    "FR" => "Espagne - Îles Canaries"  
  }
)

```
_________

```ruby
# Adding a call to update_cache will make normal data + 'IC' globally available
ISO3166::Data.update_cache


ISO3166::Data.register(
  alpha2: "IC",
  name: 'Spain - Canary Islands',
  translations: {
    'EN' => "Spain - Canary Islands",
    'DE' => "Spanien - Kanarische Inseln",
    'NL' => "Spanje - Canarische Eilanden",
    'JA' => "スペイン - カナリア諸島",
    'ES' => "España - Islas Canarias",
    "FR" => "Espagne - Îles Canaries"  
  }
)
```

Thank you !

**Comment by clm-a** [comment]
Plus, I would notify to @rposborne that translation locale keys in the docs should be downcased

``` ruby
ISO3166::Data.register(
  alpha2: "LOL",
  name: 'Happy Country'
  translations: {
    'en' => "Happy Country",
    'de' => "glückliches Land"
  }
)

ISO3166::Country.new('LOL').name = 'Happy Country'
```

**Comment by rposborne** [comment]
Should be fixed by #360 @clm-a / @madshargreave can you confirm.
